v3/internal/assetserver/webview/request_finalizer.go

```
@@ -1,34 +1,129 @@
+		nativeRequestContexts.Unlock()
+
+		// A retained native task should keep its address unique for its lifetime.
+		// If a platform nevertheless reuses an address before cleanup, retire the
+		// old context without letting its eventual Close remove the replacement.
+		if previous.cancel != nil {
+			previous.cancel()
+		}
+	}
 	// Make sure to async release since it might block the finalizer goroutine for a longer period
-	runtime.SetFinalizer(rf, func(obj *requestFinalizer) { rf.close(true) })
+	runtime.SetFinalizer(rf, func(obj *requestFinalizer) { obj.close(true) })
```

golangci-lint (2.12.2)

[error] Error return value of `obj.close` is not checked

(errcheck)